pkg/controller/secretannotator/secretannotator_controller.go

```diff
@@ -17,199 +17,32 @@ limitations under the License.
-	AzureTenantID       = "azure_tenant_id"
+	"github.com/openshift/cloud-credential-operator/pkg/controller/platform"
+	"github.com/openshift/cloud-credential-operator/pkg/controller/secretannotator/aws"
+	"github.com/openshift/cloud-credential-operator/pkg/controller/secretannotator/azure"
+	log "github.com/sirupsen/logrus"
 )
 
 func Add(mgr manager.Manager) error {
-	return add(mgr, newReconciler(mgr))
-}
-
-func newReconciler(mgr manager.Manager) reconcile.Reconciler {
-	return &ReconcileCloudCredSecret{
-		Client:           mgr.GetClient(),
-		logger:           log.WithField("controller", controllerName),
-		AWSClientBuilder: ccaws.NewClient,
-	}
-}
-
-func cloudCredSecretObjectCheck(secret metav1.Object) bool {
-	if secret.GetNamespace() == CloudCredSecretNamespace && secret.GetName() == CloudCredSecretName {
-		return true
-	}
-	return false
-}
-
-func add(mgr manager.Manager, r reconcile.Reconciler) error {
-	// Create a new controller
-	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	platformType, err := platform.Get(mgr)
+	log.Infof("Setting up secret annotator. Platform Type is %s", platformType)
 	if err != nil {
-		return err
+		log.Fatal(err)
 	}
 
-	// Watch for changes to cluster cloud secret
-	p := predicate.Funcs{
-		UpdateFunc: func(e event.UpdateEvent) bool {
-			return cloudCredSecretObjectCheck(e.MetaNew)
-		},
-		CreateFunc: func(e event.CreateEvent) bool {
-			return cloudCredSecretObjectCheck(e.Meta)
-		},
-		DeleteFunc: func(e event.DeleteEvent) bool {
-			return cloudCredSecretObjectCheck(e.Meta)
-		},
-	}
-	err = c.Watch(&source.Kind{Type: &corev1.Secret{}}, &handler.EnqueueRequestForObject{}, p)
-	if err != nil {
-		return err
-	}
-	return nil
-}
-
-var _ reconcile.Reconciler = &ReconcileCloudCredSecret{}
-
-type ReconcileCloudCredSecret struct {
-	client.Client
-	logger           log.FieldLogger
-	AWSClientBuilder func(accessKeyID, secretAccessKey []byte, infraName string) (ccaws.Client, error)
-}
-
-// Reconcile will annotate the cloud cred secret to indicate the capabilities of the cred's capabilities:
-// 1) 'mint' for indicating that the creds can be used to create new sub-creds
-// 2) 'passthrough' for indicating that the creds are capable enough for other components to reuse the creds as-is
-// 3) 'insufficient' for indicating that the creds are not usable for the cluster
-// +kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch;update
-func (r *ReconcileCloudCredSecret) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	r.logger.Info("validating cloud cred secret")
-
-	secret := &corev1.Secret{}
-	err := r.Get(context.Background(), request.NamespacedName, secret)
-	if err != nil {
-		r.logger.Debugf("secret not found: %v", err)
-		return reconcile.Result{}, err
+	switch platformType {
+	case configv1.AzurePlatformType:
+		return azure.Add(mgr, azure.NewReconciler(mgr))
+	case configv1.AWSPlatformType:
```

Comment:
go doesn't have automatic fallthroughs. need to put an explicit 'fallthrough' statement. right now on aws i get 
```
INFO[0001] setting up controller                        
INFO[0003] Setting up secret annotator. Platform Type is AWS 
ERRO[0003] platform not supportedunable to register controllers to the manager 
exit status 1
```

Reply:
ah thanks

Reply:
Not fond of the `fallthrough`keyword, I explicitely call the aws one in both cases instead